test/suites/gpu/suite_test.go

```diff
@@ -51,21 +51,32 @@ var _ = AfterEach(func() { env.Cleanup() })
 var _ = AfterEach(func() { env.AfterEach() })
 
 var _ = Describe("GPU", func() {
-	// Table test for gpu passing in different node classes
+	// Table test for GPU provisioning
 	DescribeTable("should provision one GPU node and one GPU Pod",
 		func(nodeClass *v1alpha2.AKSNodeClass) {
+			// Enable NodeRepair feature gate if running in-cluster
+			if env.InClusterController {
+				// One scenario we want to test, is that our node-auto-repair isn't too aggressive.
+				// If a gpu vm joins the cluster at 3 minutes, and takes 10 minutes to get ready, we don't want
+				// to GC the vm for node auto repair.
+				// By enabling the controller, the below assertions that the gpu workload eventually gets scheduled
+				// and runs means we didnt remove the not-ready vm
+				env.ExpectSettingsOverridden(corev1.EnvVar{Name: "FEATURE_GATES", Value: "NodeRepair=True"})
+			}
+
 			nodePool := env.DefaultNodePool(nodeClass)
 
-			// Relax default SKU family selector to allow for GPU nodes
+			// Relax SKU family selector to allow GPU SKUs
 			test.ReplaceRequirements(nodePool, karpv1.NodeSelectorRequirementWithMinValues{
-				NodeSelectorRequirement: v1.NodeSelectorRequirement{
+				NodeSelectorRequirement: corev1.NodeSelectorRequirement{
 					Key:      v1alpha2.LabelSKUFamily,
-					Operator: v1.NodeSelectorOpExists,
-				}})
-			// Exclude some of the more expensive GPU SKUs
+					Operator: corev1.NodeSelectorOpExists,
+				},
+			})
+
 			nodePool.Spec.Limits = karpv1.Limits{
-				v1.ResourceCPU:                    resource.MustParse("25"),
-				v1.ResourceName("nvidia.com/gpu"): resource.MustParse("1"),
+				corev1.ResourceCPU:                    resource.MustParse("25"),
+				corev1.ResourceName("nvidia.com/gpu"): resource.MustParse("1"),
```

Comment:
While its good we limit some of the more expensive skus, it also means a test like this one where we add node auto repair to gpus only run on smaller ones. 

Maybe the larger ones take longer to go from unready -> ready